actors/util/actor_util.go

```diff
@@ -67,6 +67,7 @@ type MinerSetHAMT map[addr.Address]bool
 type ActorIDSetHAMT map[abi.ActorID]bool
 
 type MinerEvent struct {
+	EventType abi.CronEventType
```

Comment:
This event type is a parochial actor-specific issue leaking into the high levels. As is the sectors array below.

I think a superior pattern would be to replace both of those fields with a `[]byte` and let individual actors serialize something into those bytes as they wish (or not). Then the cron actor won't be burdened with unwanted knowledge about actor internals.

Reply:
So it's not quite as leaky as you think.

Specifically, this only applies to StorageMinerActor-triggered cron events. But the SMA relies on the SPA for callbacks here and never directly communicates with the CronActor. The SPA has a callback at every tick, and then uses these callback requests (and provided info) from the SMA to callback the appropriate miner actor as needed.

Changed nonetheless, and refactored to avoid noop checks with eventTypes